tools/migrate-devspaces-assets.sh

```diff
@@ -125,11 +133,25 @@ else
    echo "Please log in or make sure that you can push images to '$CONTAINERREGISTRY' container registry."
 fi
 
-echo "docker build - '$PROJECTNAME'"
-docker build -f "$PROJECTROOT/Dockerfile.develop" -t "$CONTAINERREGISTRY/$IMAGENAMEANDTAG" "$PROJECTROOT"
-echo
-echo "Pushing image: $CONTAINERREGISTRY/$IMAGENAMEANDTAG"
-docker push "$CONTAINERREGISTRY/$IMAGENAMEANDTAG"
+# Check if running on azure cloud shell
+if [ command -v clouddrive &> /dev/null ]; then
+   CLOUDDRIVEHELP=$(clouddrive -h)
+   if [[ $CLOUDDRIVEHELP == *"Azure Cloud Shell"* ]]; then
+      ISAZURESHELL="true"
+      echo "Detected Azure Cloud Shell."
+   fi
+fi
+
+if [[ "$ISAZURESHELL" == "true" ]]; then
+   echo "Building and pushing to Azure Container registry..."
```

Comment:
Would there be a way to enforce that the container registry the user targets is ACR? If it doesn't, what error message do we provide to the user?

Reply:
The az acr build task will fail with an error message saying that no azure container registry has been found, if the user has passed in a dockerhub container registry.
Also the docs, readme and the help text is updated to mention this clearly that on azure cloud shell, only ACR is supported.